elyra/pipeline/registry.py

```diff
@@ -0,0 +1,483 @@
+            new_parameter, parameter_info = self.build_parameter(output_object, "output")
+
+            # TODO Adjust this to return an empty value for whatever type the parameter is?
+            default_value = ""
+
+            # Add to existing parameter list
+            component_parameters['parameters'].append(new_parameter)
+            component_parameters['current_parameters'][new_parameter['id']] = default_value
+
+            # Add to existing parameter info list
+            component_parameters['uihints']['parameter_info'].append(parameter_info)
+
+            # Add parameter to output group info
+            output_group_info['parameter_refs'].append(new_parameter['id'])
+
+        # Append output group info to parameter details
+        component_parameters['uihints']['group_info'][0]['group_info'].append(output_group_info)
+
+        return component_parameters
+
+    def build_parameter(self, obj, obj_type):
+        new_parameter = {}
+
+        new_parameter['id'] = f"elyra{obj_type}_{obj['name'].replace(' ', '_')}"
+
+        # Assign type, default to string??
+        if "type" in obj:
+            new_parameter['type'] = obj['type']
+        else:
+            new_parameter['type'] = "string"
+
+        # Determine whether parameter is optional
+        if ("optional" in obj and obj['optional']) \
+                or ("description" in obj and "required" in obj['description'].lower()):
+            new_parameter['required'] = True
+        else:
+            new_parameter['required'] = False
+
+        # Build parameter_info
+        # TODO Determine if any other param info should be added here, e.g. control, separator, orientation, etc.
+        parameter_info = {
+            'parameter_ref': new_parameter['id'],
+            'label': {
+                "default": obj['name']
+            },
+            'description': {
+                "default": obj['description']
+            }
+        }
+
+        # TODO Add conditions?
+
+        return new_parameter, parameter_info
+
+    def parse_component_execution_instructions(self, component_body):
+        pass
+
+
+class AirflowComponentParser(ComponentParser):
+    _type = "airflow"
+
+    def __init__(self):
+        super().__init__()
+
+    def parse_component_details(self, component_body, component_name):
+        # TODO: Maybe use this loop to get description?
+        for line in component_body:
+            print(line)
+
+        label = ' '.join(component_name.split('_')).title()
+        description = ""
+
+        component_json = {
+            'label': label,
+            'image': "",
+            'id': get_id_from_name(label),
+            'description': ' '.join(description.split()),
+            'node_types': []
+        }
+
+        node_type = set_node_type_data(get_id_from_name(label), label, description)
+        component_json['node_types'].append(node_type)
+
+        return component_json
+
+    def parse_component_properties(self, component):
```

Comment:
Please remove if this won't be necessary and update superclass to `pass` or `return None` - same difference.

Reply:
Working on implementing this this week 😄